server/modules/surveyRdb/repository/dataView/read.js

```diff
@@ -12,31 +12,133 @@ import * as Expression from '../../../../../core/expressionParser/expression'
 
 import * as DataCol from '../../schemaRdb/dataCol'
 import * as DataTable from '../../schemaRdb/dataTable'
 
-export const runSelect = async (
-  surveyId,
-  cycle,
-  tableName,
-  cols,
-  offset,
-  limit,
-  filterExpr,
-  sort = [],
-  queryStream = false,
-  client = db
-) => {
-  const schemaName = SchemaRdb.getName(surveyId)
-  // Columns
-  const colParams = cols.reduce((params, col, i) => ({ ...params, [`col_${i}`]: col }), {})
-  const colParamNames = Object.keys(colParams).map((n) => `$/${n}:name/`)
+// TODO move it to ViewDataNodeDef ?
+const _getParentNodeUuidColName = (viewDataNodeDef, nodeDef) => {
+  const { survey } = viewDataNodeDef
+  const nodeDefParent = Survey.getNodeDefParent(nodeDef)(survey)
+  return R.head(new ColumnNodeDef(viewDataNodeDef, nodeDefParent).names)
+}
+
+const _getEditModeSelectFields = (nodeDefCols) => [
+  // Node (every node is transformed into json in a column named with the nodeDefUuid)
+  ...nodeDefCols.map((nodeDefCol, idx) => `row_to_json(n${idx + 1}.*) AS "${NodeDef.getUuid(nodeDefCol)}"`),
+  // Record table fields
+  'row_to_json(r.*) AS record',
+]
+
+const _getEditModeJoins = (viewDataNodeDef, nodeDefCols) => {
+  const { surveyId } = viewDataNodeDef
+  return [
+    // Node table; one join per column def
+    ...nodeDefCols.map((nodeDefCol, idx) => {
+      const nodeDefParentUuidColName = _getParentNodeUuidColName(viewDataNodeDef, nodeDefCol)
+      const nodeDefUuid = NodeDef.getUuid(nodeDefCol)
+      const tableNode = NodeDef.isAnalysis(nodeDefCol) ? new TableResultNode(surveyId) : new TableNode(surveyId)
+      tableNode.alias = `n${idx + 1}`
+
+      return `LEFT JOIN LATERAL ( 
+        ${tableNode.getSelect({ parentUuid: `${viewDataNodeDef.alias}.${nodeDefParentUuidColName}`, nodeDefUuid })}
+      ) AS ${tableNode.alias} ON TRUE`
+    }),
+    // Record table
+    `LEFT JOIN ${getSurveyDBSchema(surveyId)}.record r 
+    ON r.uuid = ${viewDataNodeDef.columnRecordUuid}
+  `,
+  ]
+}
+
+const _dbTransformCallbackSelect = (viewDataNodeDef, editMode, nodeDefCols) => (row) => {
+  const rowUpdated = { ...row }
+  if (editMode) {
+    // Node columns (one column for each selected node def)
+    rowUpdated.cols = {}
+    nodeDefCols.forEach((nodeDefCol) => {
+      const nodeDefColUuid = NodeDef.getUuid(nodeDefCol)
+      const nodeJson = rowUpdated[nodeDefColUuid]
+      const nodeDefParentColumnUuid = _getParentNodeUuidColName(viewDataNodeDef, nodeDefCol)
+      const parentUuid = rowUpdated[nodeDefParentColumnUuid]
+      rowUpdated.cols[nodeDefColUuid] = {
+        node: dbTransformCallbackNode(nodeJson),
+        parentUuid,
+      }
+      delete rowUpdated[nodeDefColUuid]
+    })
+    // Record column
+    rowUpdated.record = camelize(rowUpdated.record)
+    // Parent node uuid column
+    const nodeDefParentColumnUuid = _getParentNodeUuidColName(viewDataNodeDef, viewDataNodeDef.nodeDef)
+    rowUpdated.parentNodeUuid = rowUpdated[nodeDefParentColumnUuid]
```

Comment:
why parentNodeUuid custom name?

Reply:
it's used to easily get the parent node uuid when generating the record edit url

Reply:
the question was, since we always use parentUuid, the custom name parentNodeUuid?

Reply:
aaa ok, it's clear now, I'll change it